packages/web-sdk/src/instrumentations/performance/navigation.ts

```diff
@@ -0,0 +1,48 @@
+import { genShortID } from '@grafana/faro-core';
+import type { EventsAPI } from '@grafana/faro-core';
+
+import { getItem, setItem, webStorageType } from '../../utils';
+
+import { NAVIGATION_ENTRY, NAVIGATION_ID_STORAGE_KEY } from './performanceConstants';
+import { calculateFaroNavigationTiming, calculateFaroResourceTiming, entryUrlIsIgnored } from './performanceUtils';
+import type { FaroNavigationItem } from './types';
+
+export function observeAndGetNavigationTimings(
```

Comment:
I think the name is a bit confusing here as this will only be run once and it acts as a Promise while hiding the observer underneath. I would suggest either renaming it to `getNavigationTimings` or expose it as an observer and let the caller use it accordingly (I prefer the first case)